cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -4727,20 +4735,38 @@ macro(build_awssdk)
     set(AWSSDK_BUILD_TYPE release)
   endif()
 
+  # provide hint for AWS SDK to link with the already located openssl
+  get_filename_component(OPENSSL_ROOT_HINT "${OPENSSL_INCLUDE_DIR}" DIRECTORY)
+  if(APPLE AND NOT OPENSSL_ROOT_HINT)
+    find_program(BREW brew)
+    if(BREW)
+      execute_process(COMMAND ${BREW} --prefix "openssl@1.1"
+                      OUTPUT_VARIABLE OPENSSL11_BREW_PREFIX
+                      OUTPUT_STRIP_TRAILING_WHITESPACE)
+      if(OPENSSL11_BREW_PREFIX)
+        set(OPENSSL_ROOT_HINT ${OPENSSL11_BREW_PREFIX})
+      else()
+        execute_process(COMMAND ${BREW} --prefix "openssl"
+                        OUTPUT_VARIABLE OPENSSL_BREW_PREFIX
+                        OUTPUT_STRIP_TRAILING_WHITESPACE)
+        if(OPENSSL_BREW_PREFIX)
+          set(OPENSSL_ROOT_HINT ${OPENSSL_BREW_PREFIX})
+        endif()
+      endif()
+    endif()
+  endif()
```

Comment:
Do we need this?
`OPENSSL_INCLUDE_DIR` must be set by `find_package(OpenSSL)`.

Reply:
I copied it from there: https://github.com/apache/arrow/pull/34159/files#diff-5cdc95f4e1b618f2f3ef10d370ce05a1ac05d9d401aecff3ccbb3d76bd366b6aL4831. Was it needed in the previous PR?

Reply:
aws-c-cal will do a find_package(OpenSSL), but i'm not sure if it needs this though

Reply:
It was needed but I should have reused `OPENSSL_ROOT_HINT`. I didn't notice it. Sorry.
Could you try removing this?

Reply:
Removed.